lib/derive-idol-err/src/lib.rs

```diff
@@ -28,25 +28,62 @@ pub fn derive(input: TokenStream) -> TokenStream {
         }
```

Comment:
Can't see it in the diff, but one line above this is

```rust
panic!("IdolError can only be derived on enums")
```

Would that have a nicer error if returned as a `compile_error` like the new ones below?

Reply:
It would; I initially didn't do that because it seemed hard, but it was not hard. Fixed.